### PR TITLE
add user type for google auth, fix signin bug

### LINKED ISSUE #128: Identify user type for Google auth

### PR DESCRIPTION
**Pull request by dilarafirtina**
fix #128

### CODE REVIEW

File: src/middlewares/passport.js [hunk 1]
```diff
@@ -1,31 +1,34 @@
 const passport = require('passport');
 const GoogleStrategy = require('passport-google-oauth20').Strategy;
 
-const { Volunteer } = require('../models/user');
+const { User } = require('../models/user');
```
Comment (yamanrajab90) [inline src/middlewares/passport.js:4]: Great work!

File: src/middlewares/passport.js [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
       clientSecret: process.env.GAPP_CLIENT_SECRET,
       callbackURL: process.env.GAPP_CLIENT_CALLBACK,
+      passReqToCallback: true,
     },
-    async (accessToken, refreshToken, profile, callBack) => {
-      const currentUser = await Volunteer.findOne({ providerId: profile.id });
-
+    async (req, accessToken, refreshToken, profile, callBack) => {
+      const currentUser = await User.findOne({
+        providerId: `google-${profile.id}`,
```
Comment (yamanrajab90) [inline src/middlewares/passport.js:16]: good catch! it query  the whole  user model.

File: src/middlewares/passport.js [hunk 3]
```diff
@@ -1,31 +1,34 @@
...
         callBack(null, currentUser);
       } else {
-        const newVolunteer = new Volunteer({
+        const user = new User({
           email: profile.emails[0].value,
           firstName: profile.name.givenName,
           lastName: profile.name.familyName,
           provider: 'Google',
           providerId: `google-${profile.id}`,
+          username: profile.emails[0].value,
```
Comment (yamanrajab90) [inline src/middlewares/passport.js:27]: Awesome work!

**Review by yamanrajab90 — APPROVED** [review]

**Review by muslimomar — APPROVED** [review]
Outstanding work

**Pull request merged**